File: 0739-daily-temperatures/0739-daily-temperatures.cpp
#include <vector>
#include <stack>
using namespace std;

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        stack<int> stack;
        int n= temp.size();
        vector<int> res(n, 0);
        for( int i =0; i<n; i++){
            while(stack.size()>0 && temp[i]> temp[stack.top()]){
                int index = stack.top();
                stack.pop();
                res[index] = i-index;
            }
            stack.push(i);
        }
        return res;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, nums: List[int]) -> List[int]:
        n=len(nums)
        res = [0]*n
        stack =[]
        for i in range(n):
            while stack and nums[i] > nums[stack[-1]]:
                prev_index = stack.pop()
                res[prev_index]  = i - prev_index
            stack.append(i)
        return res